File: grocery-comparator-mobile/services/apiService.js
// grocery-comparator-mobile/services/apiService.js
import axios from 'axios';

const BASE_URL = 'http://your-backend-api-url.com';  // Replace with your backend URL

// Function to get all products
const getProducts = () => {
  return axios.get(`${BASE_URL}/api/products`);
};

// Function to get details for a specific product
const getProductDetails = (productId) => {
  return axios.get(`${BASE_URL}/api/products/${productId}`);
};

// Function to get basket items (if needed)
const getBasket = () => {
  return axios.get(`${BASE_URL}/api/basket`);
};

export default { getProducts, getProductDetails, getBasket };

File: grocery-comparator-python-app/grocery-comparator-python-gui/api_service.py
# grocery-comparator-python-gui/api_service.py
import requests

BASE_URL = 'http://your-backend-api-url.com'  # Replace with your backend API URL

# Fetch products from the backend API
def get_products():
    response = requests.get(f'{BASE_URL}/api/products')
    return response.json() if response.status_code == 200 else []

# Get product details
def get_product_details(product_id):
    response = requests.get(f'{BASE_URL}/api/products/{product_id}')
    return response.json() if response.status_code == 200 else {}

# Fetch the current user's basket (if needed)
def get_basket():
    response = requests.get(f'{BASE_URL}/api/basket')
    return response.json() if response.status_code == 200 else []
